free/src/main/scala/cats/free/InvariantCoyoneda.scala

```diff
@@ -0,0 +1,108 @@
+package cats
+package free
+
+/**
+ * The free invariant functor on `F`. This is isomorphic to `F` as long as `F` itself is a
+ * invariant functor. The function from `F[A]` to `InvariantCoyoneda[F,A]` exists even when
+ * `F` is not an invariant functor. Implemented using a List of functions for stack-safety.
+ */
+sealed abstract class InvariantCoyoneda[F[_], A] extends Serializable { self =>
+  import InvariantCoyoneda.{unsafeApply, Aux}
+
+  /**
+   * The pivot between `fi` and `k`, usually existential.
+   */
+  type Pivot
+
+  /**
+   * The underlying value.
+   */
+  val fi: F[Pivot]
+
+  /**
+   * The list of the `Pivot => *` halves of the transformer functions, to be composed and lifted into `F` by `run`.
+   */
+  private[cats] val ks0: List[Any => Any]
```

Comment:
Do we not want to use AndThen here so we know the function composition is safe (since we control it)?

Reply:
I think so too, I was just following the style of the other `Coyoneda`s. Should I make the change across the board?